app/views/webhooks/_form.html.erb

```diff
@@ -29,5 +29,9 @@
     size: 'col-md-8', rows: 10,
     placeholder: _("Optional CAs in PEM format concatenated to verify the receiver's SSL certificate.") %>
 
+  <%= textarea_f f, :http_headers, label: _('Optional HTTP headers as JSON (ERB allowed)'),
+    size: 'col-md-8', rows: 6,
+    placeholder: "{\n\"X-Shellhook-Arg-1\": \"value\"\n}" %>
+
```

Comment:
As you saw on my yesterday's demo, this is a WIP UI, it shows a textfield with JSON, for example:

```json
{
"Argument1": "Value"
}
```

This needs to be added to the React UI PR (or possibly a separate one). I think we need to create a new React component that would allow nicer editing. The only thing to keep in mind is that some characters needs to be escaped (e.g. `"`). I think some kind of array of key-value fields with plus and minus signs would do the job, @MariaAga ?

Reply:
I'm not sure exactly what your asking here, but I think this field can be using Foremans `EditorView` (like the one in report templates) with json as a mode for nice parsing and highlighting

Reply:
Worth trying what @MariaAga suggested, maybe it will do :)

For now we could leave it as simple textarea and wait for new `new` page with `EditorView` or you could try to mount this React component into erb to see if we actually need to create a new one.